sql: Forbid setting  of zone config for non physical tables

naivcit opened the pull request:
Fixes #57478
Release note (sql change): Setting of zone configs for non physical tables is
now forbidden .

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/60592)
<!-- Reviewable:end -->

naivcit commented:
cc @rafiss

otan commented on pkg/sql/set_zone_config.go:
```diff
@@ -342,6 +342,12 @@ func (n *setZoneConfigNode) startExec(params runParams) error {
 	}
 
+	// If the table descriptor is resolved but is not a
+	// physical table(stored in the kv layer) then return an error
```
can you make this:
```
	// If the table descriptor is resolved but is not a
	// physical table then return an error.
```

otan commented on pkg/sql/set_zone_config.go:
```diff
@@ -342,6 +342,12 @@ func (n *setZoneConfigNode) startExec(params runParams) error {
+	// If the table descriptor is resolved but is not a
+	// physical table(stored in the kv layer) then return an error
+	if table != nil && !table.IsPhysicalTable() {
+		return pgerror.Newf(pgcode.WrongObjectType, `"%v" is not a physical table`, table.GetName())
```
can you make this 
```
		return pgerror.Newf(
   pgcode.WrongObjectType,
   "cannot set a zone configuration on virtual table %s",
   table.GetName(),
)
```

otan reviewed (commented):
Thanks! I have some improvements to the rror messaging.

naivcit committed:
sql: Forbid setting of zone config for non physical tables

Fixes #57478.

Release note (sql change): Setting of zone configs for non physical tables is
now forbidden .

otan commented on pkg/sql/set_zone_config.go:
```diff
@@ -342,6 +342,12 @@ func (n *setZoneConfigNode) startExec(params runParams) error {
 	}
 
+	// If the table descriptor is resolved but is not a
+	// physical table then return an error
```
super nit: this needs a full stop (.) at the end!

otan reviewed (approved):
thanks!

bors r+

naivcit commented:
The test **TestDropPhysicalTableGC** involves setting up zone config for a view and since we are restricting that here so I think the test case is failing . What should we do here ? cc @otan

otan commented:
i've pushed a fix. we should not apply the zone config the test attempts to set up if it's a view!

otan commented:
bors r+
